### DISCUSSION TITLE
loading.tsx and "use client" components

### DISCUSSION THREAD

**Discussion by ahammer-sc** [vercel/next.js#58284] [App Router]
hello, I have a loading.tsx that has a skeleton that imports some components that are client components (antd Button).

I would like to use this skeleton as my fallback loading ui, but unfortunately, the components in there require me to use "use client". Can i use a "use client" for a loading.tsx? doesn't that defeat the purpose? I wish it could be a server component, but antd Button uses context and various other client side components, so I cant render a loading skeleton without an error about createContext.

If I change it to "use client" as it wants me to, then the loading skeleton takes too long to load.

How can I accomplish a loading.tsx fallback for a route that is a client component?

**Comment by kuldeep-dhaka** [comment]
you could use something like this
<img width="890" alt="Screenshot 2023-11-22 at 6 34 32 PM" src="https://github.com/vercel/next.js/assets/106139315/ce22772d-d096-4075-a041-734e61b03f87">

**Comment by austinh** [comment]
Unfortunately that wont solve my problem. Because my loading skeleton is importing components (like antd) that use createContext, and this will throw an error when server side rendered.

**Reply by kuldeep-dhaka** [reply]
it is better to use something like shadcn/ui, this issue will persist, but you can try and create that component elsewhere make that a client component and import in your loading.tsx/loading.js
